packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -191,22 +188,82 @@ export interface ContainerRuntimeMessage {
     contents: any;
     type: ContainerMessageType;
 }
+export interface ISummaryConfigurationBaseSettings {
+    /**
+     *  Delay before first attempt to spawn summarizing container.
+     * */
+    initialSummarizerDelayMs: number;
+
+    /**
+     * Flag that will enable changing elected summarizer client after maxOpsSinceLastSummary.
+     * This defaults to false (disabled) and must be explicitly set to true to enable.
+     */
+    summarizerClientElection: boolean;
+
+    /**
+     * Defines the maximum allowed time to wait for a pending summary ack.
+     * The maximum amount of time client will wait for a summarize is the minimum of
+     * maxSummarizeAckWaitTime ( currently 10 minutes (10 * 60 * 1000) and maxAckWaitTime.
+     */
+    maxAckWaitTime: number;
+    /**
+     * Defines the maximum number of Ops in between Summaries that can be
+     * allowed before forcibly electing a new summarizer client.
+     */
+    maxOpsSinceLastSummary: number;
+}
+
+export interface ISummaryConfigurationHeuristicSettings extends ISummaryConfigurationBaseSettings {
+    /**
+     * Defines the maximum allowed time in between summarizations.
+     */
+     idleTime: number;
+    /**
+     * Defines the maximum allowed time, since the last received Ack,  before running the summary
+     * with reason maxTime.
+     */
+    maxTime: number;
+    /**
+     * Defines the maximum number of Ops, since the last received Ack, that can be allowed
+     * before running the summary with reason maxOps.
+     */
+     maxOps: number;
+    /**
+     * Defnines the minimum number of Ops, since the last received Ack, that can be allowed
+     * before running the last summary.
+     */
+    minOpsForAttemptOnClose: number;
```

Comment:
This name does not seem right. This is the min ops that will lead to a last summary but the name is totally different. Maybe `minOpsForLastSummaryAttempt`?